cmd/distrogen/distribution.go

```diff
@@ -60,8 +61,19 @@ type DistributionSpec struct {
 	Components                  *DistributionComponents `yaml:"components"`
 	Replaces                    ComponentReplaces       `yaml:"replaces,omitempty"`
 	CustomValues                map[string]any          `yaml:"custom_values,omitempty"`
-	FeatureGates                FeatureGates            `yaml:"feature_gates"`
+	FeatureGates                FeatureGates            `yaml:"feature_gates,omitempty"`
 	GoProxy                     string                  `yaml:"go_proxy,omitempty"`
+	ComponentModuleBase         string                  `yaml:"component_module_base"`
+	DistrogenVersion            string                  `yaml:"distrogen_version"`
+}
+
+// GetGoMajorVersion will parse the GoVersion in the spec and return a version without a patch
```

Comment:
I have a light convention which I did not write down or tell anyone which is that functions intended to be used in templates should be named `RenderX`. Could we change the name of this to `RenderGoMajorVersion`?